### PR TITLE
Update kubectl-gs template catalog with new CR definition

### PR DESCRIPTION
**Pull request by kubasobon**
Towards https://github.com/giantswarm/giantswarm/issues/21782

Team Honeybadger is extending Catalog CRs in order to provide Helm Catalog redundancy and allow internal catalogs to rely on OCI Registries. Where previously Catalog CRs allowed for specifying just one backend in `.spec.storage`, we now support an array of backends in `.spec.repositories`. `.spec.storage` is kept for backwards compatiblilty until older versions of app-operator are phased out. This required extending the `kubectl gs template catalog` command.

Docs https://github.com/giantswarm/docs/pull/1466

---

As the creator of a pull request, please consider:

- [x] Describe the goal you are trying to accomplish here, above the line.
- [x] Add a user-friendly description of your change to `CHANGELOG.md`.
- [x] Provide a link to the issue you are solving or working towards, if available.
- [x] Request SIG UX for review. They care about almost all user-facing changes.
- [x] Update the public [kubectl-gs documentation](https://docs.giantswarm.io/ui-api/kubectl-gs/) to reflect the changes here. 
- [ ] add the `breaking-change` label to the PR if the change you are making changes the existing behaviour. Examples: removal of a flag, removal of a command, change of a default value. (Such changes should be released with a **major version** bump.)

Feel free to remove this checklist when done.

### CODE REVIEW

File: cmd/template/catalog/flag.go [hunk 1]
```diff
@@ -36,8 +38,11 @@ func (f *flag) Init(cmd *cobra.Command) {
 	cmd.Flags().StringVar(&f.Name, flagName, "", "Catalog name.")
 	cmd.Flags().StringVar(&f.Namespace, flagNamespace, "", "Namespace where the catalog will be created.")
 	cmd.Flags().StringVar(&f.Secret, flagSecret, "", "Path to a secret file.")
-	cmd.Flags().StringVar(&f.URL, flagURL, "", "Catalog storage URL.")
+	cmd.Flags().StringArrayVar(&f.URLs, flagURL, []string{}, "Catalog storage URL.")
+	cmd.Flags().StringArrayVar(&f.URLTypes, flagURLType, []string{"helm"}, "Type of catalog storage.")
 	cmd.Flags().StringVar(&f.Visibility, flagVisibility, "public", "Visibility label for whether catalog appears in the web UI.")
+
+	// TODO(kuba): add long desc. with usage example, especially --url --type --url etc.
```
Comment (MarcelMue) [inline cmd/template/catalog/flag.go:45]: TODO seems done?

File: go.mod [hunk 2]
```diff
@@ -9,7 +9,7 @@ require (
 	github.com/coreos/go-oidc/v3 v3.2.0
 	github.com/dgrijalva/jwt-go v3.2.0+incompatible
 	github.com/fatih/color v1.13.0
-	github.com/giantswarm/apiextensions-application v0.3.1
+	github.com/giantswarm/apiextensions-application v0.3.2-0.20220601125452-b940494d929e
```
Comment (MarcelMue) [inline go.mod:12]: Should this be tagged before merging?

**Review by MarcelMue — COMMENTED** [review]
Generally LGTM

**Review by kuosandys — APPROVED** [review]

File: go.mod (hunk 2 above)
Comment (kubasobon) [reply go.mod:12]: It already is, I'll bump the version.

**Pull request merged**